src/Rydia.Database/Services/SettingsService.cs

```
@@ -0,0 +1,165 @@
+namespace Rydia.Database.Services;
+
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Rydia.Database.Constants;
+using Rydia.Database.Models;
+
+/// <summary>
+/// Service for managing configuration settings in the database
+/// </summary>
+public class SettingsService(RydiaDbContext context, ILogger<SettingsService> logger) : ISettingsService
+{
+    private readonly RydiaDbContext _context = context;
+    private readonly ILogger<SettingsService> _logger = logger;
+
+    public async Task<string?> GetSettingAsync(string key)
+    {
+        if (string.IsNullOrWhiteSpace(key))
+        {
+            _logger.LogWarning("Attempted to get setting with null or empty key");
+            return null;
+        }
+
+        if (!SettingKeys.IsValidKey(key))
+        {
+            _logger.LogWarning("Attempted to get setting with invalid key: {Key}", key);
+            return null;
+        }
+
+        try
+        {
+            var setting = await _context.Settings
+                .FirstOrDefaultAsync(s => s.Key == key);
+
+            return setting?.Value;
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Error retrieving setting with key: {Key}", key);
+            return null;
+        }
+    }
+
+    public async Task<bool> SetSettingAsync(string key, string value)
+    {
+        if (string.IsNullOrWhiteSpace(key))
+        {
+            _logger.LogWarning("Attempted to set setting with null or empty key");
+            return false;
+        }
+
+        if (!SettingKeys.IsValidKey(key))
+        {
+            _logger.LogWarning("Attempted to set setting with invalid key: {Key}", key);
+            return false;
+        }
+
+        if (value == null)
+        {
+            _logger.LogWarning("Attempted to set setting with null value for key: {Key}", key);
+            return false;
+        }
+
+        try
+        {
+            var existingSetting = await _context.Settings
```

I think for this block I would prefer to fail fast and avoid deep nesting